client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,45 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to 'CONSUMPTION' based,
+     * the Reader Group needs to register with the Stream as a Subscriber
+     * and periodically publish retention StreamCuts either manually or at checkpoints.
+     * This can be done by setting retentionConfig to 'CONSUMPTION_BASED_USER_STREAMCUT' or
+     * 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * If the Stream's RetentionPolicy is not Consumption based, retentionConfig should be 'NO_IMPACT'.
+     *
+     * If the Streams' Retention Policy is set to 'CONSUMPTION', but the ReaderGroup retentionConfig is set at
+     * 'NO_IMPACT', the ReaderGroup will not get registered as a Subscriber with the Stream
+     * and retention StreamCuts cannot be published for this ReaderGroup.
+     * This will break the promise of ConsumptionBased Retention.
+     *
+     * NO_IMPACT - Read Positions of Readers do not impact Stream truncation/retention.
+     * CONSUMPTION_BASED_USER_STREAMCUT - User provides the StreamCut for truncation using Consumption Based Retention.
+     * CONSUMPTION_BASED_AT_LAST_CHECKPOINT - StreamCut corresponding to lastCompletedCheckpoint is published as truncation point for Consumption Based Retention.
```

Comment:
Not a truncation point.
Here and elsewhere don't say "for Consumption Based Retention" refer to the specific stream configuration policy. Users don't know our PDPs or internal names for things.

Reply:
I understand users are not expected to read PDPs or design documents.
However, the intent of a Javadoc is to explain how an API works and not to provide documentation for the entire feature. 
For that we need user documentation that explains how Consumption based retention needs to be configured and used in Pravega.
GH issue for the same: https://github.com/pravega/pravega/issues/5350

Reply:
FYI those docs are in the repo https://github.com/pravega/pravega/tree/master/documentation/src/docs
so you can edit them as part of a PR, or in a branch.